### PR TITLE
Add hops limit docs

### PR DESCRIPTION
**Pull request by DavIvek**
### Description

Added the docs for the hops limit feature.

### Pull request type

Please check what kind of PR this is:

- [ ] Fix or improvement of an existing page
- [x] New documentation page, release related

### Related PRs and issues

PR this doc page is related to: 
https://github.com/memgraph/memgraph/pull/1928

Closes:
(paste the link to the issue it closes)

### Checklist:

- [x] Check all content with Grammarly
- [x] Perform a self-review of my code
- [x] Make corresponding changes to the rest of the documentation (consult with the DX team)
- [x] The build passes locally
- [x] My changes generate no new warnings or errors
- [x] Add a corresponding label
- [x] If release-related, add a product and version label
- [x] If release-related, add release note on product PR

### CODE REVIEW

**Comment by DavIvek** [comment]
Should the part about the hops limit be placed before the Weighted Shortest Path documentation (since it isn't usable with WSP or all shortest paths), at the end of the page, or on its own page? What are your thoughts?

File: pages/advanced-algorithms/deep-path-traversal.mdx [hunk 1]
```diff
@@ -346,6 +346,110 @@ MATCH path=(n {id: 0})-[*BFS (r, n, p | r.eu_border = false AND n.drinks_USD < 1
...
+```
+
+### Implementation Variations
+
+<Callout type="warning">
+
+Keep in mind number of hops throughout the query could be different in different algorithms, due to differences in the implementation of different operators.
+Also, Memgraph uses two different operators for bfs due to optimization reasons, and even with them, the number of hops can differ for the same paths.
+
+</Callout>
```
Comment (kgolubic) [inline pages/advanced-algorithms/deep-path-traversal.mdx:381]: Remove  `type="warning"` from `Callout`. It is not needed in this case.

Remove word `Also`.

File: pages/advanced-algorithms/deep-path-traversal.mdx [hunk 2]
```diff
@@ -346,6 +346,110 @@ MATCH path=(n {id: 0})-[*BFS (r, n, p | r.eu_border = false AND n.drinks_USD < 1
...
+
+### Overview
+Breadth-First Search (BFS), Depth-First Search (DFS), and expand query traversals can now be controlled by the number of hops (edges crossed). 
+Each unit of traversal, including filtering on edge types, edge properties, or visiting already processed nodes, is counted as a hop.
+
+<Callout type="warning">
+
+The hops limit does not affect the Weighted Shortest Path, All Shortest Paths, and expansions where relationships are created (CreateExpand operator).
+
+</Callout>
```
Comment (kgolubic) [inline pages/advanced-algorithms/deep-path-traversal.mdx:359]: Remove  `type="warning"` from `Callout`. It is not needed in this case.

**Review by kgolubic — APPROVED** [review]
I've left to comments. Everything else look ok.

**Review by hal-eisen-MG — APPROVED** [review]
LGTM

**Pull request merged**